samples/fhostlookup/Generator.cpp

```diff
@@ -14,49 +14,96 @@
         return false;
     }
-
     if (!configuration["database"].IsString()) {
         DARWIN_LOG_CRITICAL("HostLookup:: Generator:: \"database\" needs to be a string");
         return false;
     }
-
     db = configuration["database"].GetString();
     std::ifstream file(db.c_str());
-    std::string host;
-
     if (!file) {
         DARWIN_LOG_CRITICAL("HostLookup:: Generator:: Configure:: Cannot open host database");
         return false;
     }
 
-    while (!darwin::files_utils::GetLineSafe(file, host).eof()) {
-        if(file.fail() or file.bad()){
-            DARWIN_LOG_CRITICAL("HostLookup:: Generator:: Configure:: Error when reading host database");
-            return false;
-        }
-        if (!host.empty()){
-            _database.insert({host,0});
-        }
+    DARWIN_LOG_DEBUG("HostlookupGenerator:: Parsing database...");
+    rapidjson::IStreamWrapper isw(file);
+    database.ParseStream(isw);
+    if (not this->LoadDatabase(database)) {
+        file.close();
+        return false;
     }
-
     file.close();
     return true;
 }
 
+bool Generator::LoadDatabase(const rapidjson::Document& database) {
+    DARWIN_LOGGER;
+    if (not database.IsObject()) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: Database is not a JSON object");
+        return false;
+    }
+    if (not database.HasMember("feed_name") or not database["feed_name"].IsString()) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: No proper feed name provided in the database");
+        return false;
+    }
+    this->_feed_name = database["feed_name"].GetString();
+    if (not database.HasMember("data") or not database["data"].IsArray()) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: No or ill formated entries in the database");
+        return false;
+    }
+    auto entries = database["data"].GetArray();
+    if (entries.Size() == 0) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: No entry in the database. Stopping.");
+        return false;
+    }
+    for (auto& entry : entries) {
+        this->LoadEntry(entry);
+    }
+    if (this->_database.size() == 0) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: No usable entry in the database. Stopping.");
+        return false;
+    }
+    return true;
+}
+
+bool Generator::LoadEntry(const rapidjson::Value& entry) {
+    DARWIN_LOGGER;
+    int score = 100;
+    std::string sentry;
+    if (not entry.IsObject()) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: Database entry is not a JSON object. Ignoring.");
+        return false;
+    }
+    if (not entry.HasMember("entry") or not entry["entry"].IsString()) {
+        DARWIN_LOG_CRITICAL("HostlookupGenerator:: Entry is not a string. Ignoring.");
+        return false;
+    }
+    if (entry.HasMember("score") and entry["score"].IsInt()) {
+        score = entry["score"].GetInt();
+        if (score < 0 or score > 100)
+            score = 100;
```

Comment:
Shouldn't we add a Warning when a score is overrode ?

Reply:
Corrected in cb566b42bb72e0d550633bf8bcdd69e3808d5f1b